### PR TITLE
fix(ledger): add min pool cost to protocol parameters

### PR DESCRIPTION
**Pull request by wolf31o2**
## Summary by CodeRabbit

* **Improvements**
  * Preserve and expose MinPoolCost across protocol-parameter upgrades and conversions; improved mapping and validation across eras, including Alonzo upgrades.
  * Centralized and strengthened value/fee/size validation to ensure conservation, correct fee/min-UTXO calculations, and max-tx-size checks.

* **New Features**
  * New structured protocol-parameter type with CBOR-backed partial updates and reliable conversion to UTxO-era parameters, including rational-field validation.
  * Public helpers to compute minimum fee and minimum UTXO value.

* **Tests**
  * Extensive unit tests for upgrades, conversions, CBOR decoding, pointer semantics, edge cases (min/max), Alonzo-specific field handling, and error paths.

### CODE REVIEW

File: ledger/mary/pparams.go [hunk 1]
```diff
@@ -14,14 +14,197 @@
...
+		TreasuryExpansion: &cardano.RationalNumber{
+			Numerator:   int32(p.Tau.Num().Int64()),
+			Denominator: uint32(p.Tau.Denom().Int64()),
+		},
+		MinPoolCost: p.MinPoolCost,
+		ProtocolVersion: &cardano.ProtocolVersion{
+			Major: uint32(p.ProtocolMajor),
+			Minor: uint32(p.ProtocolMinor),
+		},
+	}, nil
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply ledger/mary/pparams.go]: There is no MinUtxoValue in cardano.PParams

**Comment by wolf31o2** [comment]
@coderabbitai this is failing tests due to an incorrect object type, which you did not catch in review. What is the proper fix for these test errors introduced by this pull request?

**Review by verbotenj — APPROVED** [review]
